markdown/spring-cloud-stream-beyond/spring-cloud-stream-beyond.md

```diff
@@ -0,0 +1,620 @@
+
+👀 Under the covers this option will use a Non-Durable Anonymous Queue on the broker to hold the messages for the consuming microservice. Let's check it out!
+This option will deliver events in order to your microservice while it remains up and running. To do this we just to specify a `destination`, but NO `group` on your input binding. 
+
+So open your `application.yml` file and add the following config to what already exists. This configuration is telling Spring Cloud Stream that you want your `myConsumer` 
+``` yaml
+spring:
+  cloud:
+    function:
+      definition: myConsumer
+    stream:
+      bindings:
+        myConsumer-in-0:
+          destination: spring/cloud/stream
+```
+
+Go ahead and run your app. You should see the app start up and connect to the event broker. Note that under the covers the Solace Binder will bind your function to a Non-Durable Anonymous (or Temporary) queue on the broker. 
+This endpoint type in Solace is a temporary queue that will deliver messages to your app in order while your app remains online. It however is NOT a durable endpoint and will be removed after your application goes offline for more than 30 seconds.     
+
+You'll see the name of your Anonymous Queue shown in a log entry on the console from the *SolaceQueueProvisioner* class: 
+```
+SolaceQueueProvisioner : Subscribing queue #P2P/QTMP/v:b0e95afab69a/scst/an/59e7cbe3-ec38-4da8-a8fa-d1e428f6eb56/plain/spring/cloud/stream to topic spring/cloud/stream
+
+```
+
+🛠 **Let's test this out using the *Try-Me* tab in Solace Cloud.**
+1. Navigate to the Solace Cloud Console
+1. Choose "Cluster Manager" -> and click the messaging service you created earlier
+1. Click the "Try-Me!" option on the top menu
+1. Click "Connect" on the "Publisher" side
+1. Type in the `spring/cloud/stream` topic
+1. Change the "Message" to whatever you'd like and click "Send"
+
+Positive
+: 💥 Note that you'll see that the messages arrive in the order that you send them. 
+
+### TODO - INSERT TRY-ME IMAGE.
+
+This option is great when we don't need to store messages/events when the app is offline, but what if we do!?
+
+### Publish-Subscribe (Durable)
+Sometimes it isn't enough to be able to deliver messages/events only while the app remains connected to the broker; there are many scenarios where you want messages stored while the app is offline while still maintaining order. In this case the Solace Binder allows you to follow the Publish-Subscribe pattern and consume from an Exclusive Durable Queue that will do just that. 
+
+Positive
+: 💥 As a bonus, this configuration also allows you to create a Primary/Secondary type of configuration where you can have multiple microservices ready to process the events and if the Primary disconnects the Secondary will take over. 
```

Comment:
Out of curiosity, is this leader election?

Reply:
not really, this is just maintaining message order with a durable queue. But yes, you do have a primary/secondary as well. The concept of leader election is moreso allowing your apps to use some sort of  logic to decide who the leader is to execute some other task. The leader election project that was donated to the community allows you to use a solace exclusive queue to help perform that  logic, but your app might not even be doing something messaging related. Does that make sense?